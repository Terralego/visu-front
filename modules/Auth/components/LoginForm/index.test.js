import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { LoginForm } from './';

it('should render correctly', () => {
  const tree = renderer
    .create(<LoginForm />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('should submit form with success', async done => {
  const authAction = jest.fn();
  const wrapper = shallow(<LoginForm
    authAction={authAction}
  />);

  wrapper.setState({ login: 'foo@bar', password: 'bar' });

  await wrapper.instance().submit({ preventDefault () {} });

  expect(authAction).toHaveBeenCalledWith({ login: 'foo@bar', password: 'bar' });
  expect(wrapper.state().errorLogin).toBe(false);
  expect(wrapper.state().errorPassword).toBe(false);

  done();
});

it('should submit form with failure', async done => {
  const authAction = jest.fn(() => {
    const error = new Error();
    error.data = {
      email: ['invalid'],
    };
    throw error;
  });
  const wrapper = shallow(<LoginForm
    authAction={authAction}
  />);

  await wrapper.instance().submit({ preventDefault () {} });

  expect(authAction).toHaveBeenCalledWith({ login: undefined, password: undefined });
  expect(wrapper.state().errorLogin).toBe(true);
  expect(wrapper.state().errorPassword).toBe(false);

  done();
});
