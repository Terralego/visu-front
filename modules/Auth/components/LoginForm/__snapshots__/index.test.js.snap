// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="bp3-card bp3-elevation-0 login-form"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="bp3-form-group"
    >
      <label
        className="bp3-label"
        htmlFor="login"
      >
        Email
         
        <span
          className="bp3-text-muted"
        >
          (required)
        </span>
      </label>
      <div
        className="bp3-form-content"
      >
        <div
          className="bp3-input-group"
        >
          <input
            className="bp3-input"
            id="login"
            onChange={[Function]}
            placeholder="Email"
            style={
              Object {
                "paddingRight": 10,
              }
            }
            type="text"
          />
        </div>
        <div
          className="bp3-form-helper-text"
        >
          Type your email
        </div>
      </div>
    </div>
    <div
      className="bp3-form-group"
    >
      <label
        className="bp3-label"
        htmlFor="password"
      >
        Password
         
        <span
          className="bp3-text-muted"
        >
          (required)
        </span>
      </label>
      <div
        className="bp3-form-content"
      >
        <div
          className="bp3-input-group"
        >
          <input
            className="bp3-input"
            id="password"
            onChange={[Function]}
            placeholder="Password"
            style={
              Object {
                "paddingRight": 10,
              }
            }
            type="password"
          />
        </div>
        <div
          className="bp3-form-helper-text"
        >
          Type your password
        </div>
      </div>
    </div>
    <button
      className="bp3-button"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      type="submit"
    >
      <span
        className="bp3-button-text"
      >
        Connexion
      </span>
    </button>
  </form>
</div>
`;
